### PR TITLE
Use gorm transactions w rollback for multi add/remove policies

### PR DESCRIPTION
**Pull request by dallen42**

### CODE REVIEW

File: adapter.go [hunk 1]
```diff
@@ -388,28 +388,30 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
...
-	for _, rule := range rules {
-		line := a.savePolicyLine(ptype, rule)
-		err = a.db.Create(&line).Error
-		if err != nil {
-			return err
+	return a.db.Transaction(func(tx *gorm.DB) error {
+		for _, rule := range rules {
+			line := a.savePolicyLine(ptype, rule)
+			if err := tx.Create(&line).Error; err != nil {
+				tx.Rollback()
```
Comment (rwbarber2) [inline adapter.go]: According to the docs at https://gorm.io/docs/transactions.html#Transactions the returning of an err will rollback the transaction so no need to call tx.Rollback().

File: adapter.go [hunk 2]
```diff
@@ -388,28 +388,30 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
...
-	var err error
-	for _, rule := range rules {
-		line := a.savePolicyLine(ptype, rule)
-		err = a.rawDelete(a.db, line) //can't use db.Delete as we're not using primary key http://jinzhu.me/gorm/crud.html#delete
-		if err != nil {
-			return err
+	return a.db.Transaction(func(tx *gorm.DB) error {
+		for _, rule := range rules {
+			line := a.savePolicyLine(ptype, rule)
+			if err := a.rawDelete(a.db, line); err != nil { //can't use db.Delete as we're not using primary key http://jinzhu.me/gorm/crud.html#delete
```
Comment (rwbarber2) [inline adapter.go]: Should'nt 'tx' be passed to rawDelete?

File: adapter.go [hunk 3]
```diff
@@ -388,28 +388,30 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
...
+				tx.Rollback()
```
Comment (rwbarber2) [inline adapter.go]: Not needed here

**Review by rwbarber2 — CHANGES_REQUESTED** [review]
Must act on the transaction object.

File: adapter.go (hunk 2 above)
Comment (dallen42) [reply adapter.go]: @rwbarber2 correct, fixed

File: adapter.go (hunk 1 above)
Comment (dallen42) [reply adapter.go]: @rwbarber2 fixed

File: adapter.go (hunk 3 above)
Comment (dallen42) [reply adapter.go]: @rwbarber2 fixed

**Review by rwbarber2 — APPROVED** [review]
LGTM

**Pull request merged**